readinglist/tests/test_views_article.py

```diff
@@ -85,24 +86,129 @@ def test_string_filters_naively_by_value(self):
+        self.assertEqual(len(resp.json['items']), 0)
+
+    def test_filter_from_last_header_value_is_exclusive(self):
+        article = MINIMALIST_ARTICLE.copy()
+        resp = self.app.post_json('/articles', article, headers=self.headers)
+
+        current = float(resp.headers['Timestamp'])
+        url = '/articles?_since={0:.3f}'.format(current)
+
+        resp = self.app.get(url, headers=self.headers)
+        self.assertEqual(len(resp.json['items']), 0)
+
     def test_filter_works_with_empty_list(self):
         self.fxa_verify.return_value = {
             'user': 'jean-louis'
         }
         self.app.get('/articles?unread=true', headers=self.headers)
 
+    def test_timestamp_are_always_identical_on_read(self):
+
+        def read_timestamp():
+            resp = self.app.head('/articles', headers=self.headers)
+            return int(resp.headers['Timestamp'])
+
+        before = read_timestamp()
+        now = read_timestamp()
+        after = read_timestamp()
+        self.assertTrue(before == now == after)
+
+    def test_timestamp_are_always_incremented_on_creation(self):
+
+        def read_timestamp():
+            resp = self.app.post_json('/articles',
+                                      MINIMALIST_ARTICLE,
+                                      headers=self.headers)
+            return int(resp.json['last_modified'])
+
+        before = read_timestamp()
+        now = read_timestamp()
+        after = read_timestamp()
+        self.assertTrue(before < now < after)
+
+    def test_records_created_during_fetch_are_above_fetch_revision(self):
+
+        revisions = {}
+
+        def long_fetch():
+
+            def delayed_get(*args, **kwargs):
+                time.sleep(.25)
+                return MINIMALIST_ARTICLE
+
+            with mock.patch.object(self.db, 'get_all', delayed_get):
+                resp = self.app.head('/articles',
+                                     headers=self.headers)
+                revisions['fetch'] = resp.headers['Timestamp']
+
+        thread = threading.Thread(target=long_fetch)
+        thread.start()
+        resp = self.app.post_json('/articles',
+                                  MINIMALIST_ARTICLE,
+                                  headers=self.headers)
+        revisions['post'] = resp.headers['Timestamp']
+        thread.join()
+        self.assertTrue(revisions['post'] > revisions['fetch'])
+
+    def test_timestamps_are_thread_safe(self):
+        obtained = []
+
+        def hit_post():
+            for i in range(100):
+                resp = self.app.post_json('/articles',
+                                          MINIMALIST_ARTICLE,
+                                          headers=self.headers)
+                current = int(resp.json['last_modified'])
+                obtained.append(current)
+
+        thread1 = threading.Thread(target=hit_post)
+        thread2 = threading.Thread(target=hit_post)
+        thread1.start()
```

Comment:
it might be wise to cretate a _create_thread function in your test, so you can make sure all threads are cleaned up on tearDown